### ISSUE TITLE
Apriltags-cpp should be checked out at clone, not at build time

### ISSUE THREAD

**Issue by ClintLiddick** [personalrobotics/apriltags#8]
We should switch to either git submodules, or git tree, or instructions to clone/symlink the apriltags-cpp code. The current setup makes it impossible to rebuild without an internet connection.

**Comment by psigen** [comment]
One option might be to use the following option to `ExternalProject`:
```
If UPDATE_DISCONNECTED is set, the update step is not executed automatically when building the main target. The update step can still be added as a step target and called manually. This is useful if you want to allow to build the project when you are disconnected from the network (you might still need the network for the download step). This is disabled by default. The directory property EP_UPDATE_DISCONNECTED can be used to change the default value for all the external projects in the current directory and its subdirectories.
```

**Comment by davelkan** [comment]
This is only a valid option with Cmake 3.2 or later. People using indigo for example cannot use this option. I would argue therefore that this is still an issue. There is no real reason Apriltags users should need internet to build.

**Comment by andre-nguyen** [comment]
This is indeed a problem for me since sometimes I do tests in the field without internet access. Any idea how to fix this easily?

**Comment by andre-nguyen** [comment]
I ended up doing a quick and dirty hack to catkinize https://github.com/swatbotics/apriltags-cpp and put it as a dependency of this repo. I doubt this is the direction you guys seem to be aiming for though.

**Comment by psigen** [comment]
Actually, in discussions with @mkoval, we concluded that catkin-izing the apriltags upstream library is probably exactly _The Right Thing_ for this case.

@andre-nguyen can you make a PR for the catkinization?  @ClintLiddick, if you and @mkoval can help clean it up and give it the OK, I'll merge it up there.  Then we can change this package to simply use CMake to `FindApriltags()`, and change the directions to have people check out both packages together.

This may be useful as a step towards #19 anyway.

**Comment by davelkan** [comment]
This sounds like a great plan to me.

**Comment by mkoval** [comment]
I agree with @psigen that *The Right Thing* is to build the upstream library as a separate package. All we need to do is modify the upstream repository to follow [REP-136](http://www.ros.org/reps/rep-0136.html) by:

1. adding a `package.xml` file with `<build_type>cmake</build_type>`
2. adding an `install(FILES)` rule for the `package.xml` file
3. using [`configure_package_config_file()`](https://cmake.org/cmake/help/v3.0/module/CMakePackageConfigHelpers.html) to generate a `ApriltagsConfig.cmake` file

You can see an example of this in Aikido's [`package.xml`](https://github.com/personalrobotics/aikido/blob/58c3357a84e20909d1c0c23a07c615c4aabc34d1/aikido/package.xml) and [`CMakeLists.txt`](https://github.com/personalrobotics/aikido/blob/master/aikido/CMakeLists.txt#L236) files.

Once this is done, this package can depend on Apriltags using `find_package(Apriltags)`. The two packages may be built in the same workspace using `catkin_make_isolated` or `catkin build`.

**Comment by andre-nguyen** [comment]
I see.

This isn't quite how I did it here https://github.com/swatbotics/apriltags-cpp/pull/8 but your method seems cleaner. I'll leave it up for anyone who just want's to have the quick fix.

**Comment by davelkan** [comment]
Any updates on this?

**Comment by psigen** [comment]
Not at the moment, I think everyone's been pretty busy recently.

**Comment by Asukamax** [comment]
Excuse me .I'm a newbie.
I don't know how to use this package ,even I could build it ,run the launch file successful...
When I run the apriltags.launch ,it seem like doing nothing?
Is that I should run some node to make my camera active and publish the image imformation to topic?
If is that , which package I could use?

Thanks a lot

**Comment by psigen** [comment]
@Asukamax: It's normally useful to create a new issue for different topics, so I've created one here for you #23.  Feel free to post followup questions there.

**Comment by psigen** [comment]
Fixed in #25
